src/borg/archive.py

```diff
@@ -858,10 +858,13 @@ def same_item(item, st):
             st = os.stat(path, follow_symlinks=False)
             if continue_extraction and same_item(item, st):
                 return  # done! we already have fully extracted this file in a previous run.
-            elif stat.S_ISDIR(st.st_mode):
-                os.rmdir(path)
-            else:
+            # remove anything that is not a directory
+            if not stat.S_ISDIR(st.st_mode):
                 os.unlink(path)
+            # only remove a directory if it is conflicting
+            # preserve existing directories because they might be subvolumes
+            elif not stat.S_ISDIR(item.mode):
+                os.rmdir(path)
```

Comment:
It's rather unclear still whether the resulting directory after extraction will be correct (== as in the archive) in all cases:
- simple attrs
- xattrs
- acls

The existing directory could have some metadata set already, but the resulting directory must be exactly what we have in the archived item, not a mix up of fs item and archived item.

Reply:
The code that makes sure about this could be also useful later if we implement extracting into a non-empty directory (for more than the simplest cases, like the already implemented "continue an extraction").

Reply:
hmm, thinking about it...

we currently more or less require extracting into an empty dir (exception: that "continue extraction" feature). we could also require that if there is any pre-existing directory inside the extraction directory, it must not have any special attrs set.